### PR TITLE
added parallelization in analysis.dihedrals

### LINKED ISSUE #4673: `MDAnalysis.analysis.dihedrals`: Implement parallelization or mark as unparallelizable
Given a newly implemented parallelization (https://docs.mdanalysis.org/dev/documentation_pages/analysis/parallelization.html), implement it for all classes in `analysis.dihedrals` module. You may use `analysis.RMSD` in `develop` as a reference.

You need to add:

 - appropriate implementation of result aggregation for AnalysisBase subclasses in `analysis.dihedrals` module (or explicitly limit available backends if the classes aren't parallelizable)
 - a boilerplate fixture(s) to `testsuite/analysis/conftest.py`, analogous with existing ones
 - a `client_...` fixtures to all tests using in `testsuite/MDAnalysisTests/analysis/test_dihedrals.py`, and modify the way run() method is called

### PR DESCRIPTION
**Pull request by orbeckst**
Fixes #4673

Changes made in this Pull Request:
- parallelized Dihedral, Ramachandran, Janin
- added tests (including fixtures in analysis/conftest.py)
- update CHANGELOG

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
📚 Documentation preview 📚: https://mdanalysis--4682.org.readthedocs.build/en/4682/

### CODE REVIEW

File: package/CHANGELOG [hunk 1]
```diff
@@ -17,7 +17,7 @@ The rules for this file:
 ??/??/?? IAlibay, HeetVekariya, marinegor, lilyminium, RMeli,
          ljwoods2, aditya292002, pstaerk, PicoCentauri, BFedder,
          tyler.je.reddy, SampurnaM, leonwehrhan, kainszs, orionarcher,
-         yuxuanzhuang, PythonFZ, laksh-krishna-sharma
+         yuxuanzhuang, PythonFZ, laksh-krishna-sharma, orbeckst
```
Comment (talagayev) [inline package/CHANGELOG]:
```suggestion
         yuxuanzhuang, PythonFZ, laksh-krishna-sharma, orbeckst,
         talagayev
```

File: testsuite/MDAnalysisTests/analysis/conftest.py [hunk 2]
```diff
@@ -91,11 +93,33 @@ def client_RMSF(request):
...
-  
+
+# MDAnalysis.analysis.bat
+
 @pytest.fixture(scope='module', params=params_for_cls(BAT))
 def client_BAT(request):
-  return request.param
\ No newline at end of file
+  return request.param
+
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/analysis/conftest.py]:
```suggestion
    return request.param
```

Comment (orbeckst) [reply testsuite/MDAnalysisTests/analysis/conftest.py]: Make darker happy.

[Line 124:3](https://github.com/MDAnalysis/mdanalysis/blob/7cdde28719ad26c2b9dbd7bfae5a2387934d6119/testsuite/MDAnalysisTests/analysis/conftest.py#L124): [E111](https://duckduckgo.com/?q=pep8%20E111) indentation is not a multiple of four
[Line 125:1](https://github.com/MDAnalysis/mdanalysis/blob/7cdde28719ad26c2b9dbd7bfae5a2387934d6119/testsuite/MDAnalysisTests/analysis/conftest.py#L125): [W391](https://duckduckgo.com/?q=pep8%20W391) blank line at end of file

**Comment by orbeckst** [comment]
@yuxuanzhuang could you please shepherd the PR to completion? Thanks!

(When you squash-merge, please write a nice log message such as
```
* Fixes #4673
* Changes made in this Pull Request:
    - parallelized Dihedral, Ramachandran, Janin
    - added tests (including fixtures in analysis/conftest.py)
    - update CHANGELOG

Co-authored: ...
```
and keep the co-author line(s), in particular @talagayev ! Thanks!)

**Review by yuxuanzhuang — APPROVED** [review]
lgtm! Thank @talagayev for your help as well!

**Comment by talagayev** [comment]
Nice, happy to help :)

**Pull request merged**